_posts/blog/2019-09-17-erasure-coding.md

```diff
@@ -0,0 +1,177 @@
+---
+title: "Using HDFS Erasure Coding with Accumulo"
+author: Ed Seidl
+reviewers:
+---
+
+HDFS by default uses triple replication for both performance and durability reasons.  Hadoop 3, 
+however, introduced the use of erasure coding (EC), which can improve durability while decreasing overhead.
+Since Accumulo 2.0 now directly supports Hadoop 3, it's time to take a look at whether using
+EC with Accumulo makes sense.
+
+* [EC Intro](#ec-intro)
+* [EC Performance](#ec-performance)
+* [Accumulo Performance with EC](#accumulo-performance-with-ec)
+
+### EC Intro
+
+By default HDFS file systems achieve durability via block replication.  Usually
+the replication level is set to 3, resulting in a disk overhead of 200%. Hadoop 3 
+introduced EC as a better way to achieve durability.  More info can be
+found [here](https://hadoop.apache.org/docs/r3.2.0/hadoop-project-dist/hadoop-hdfs/HDFSErasureCoding.html).
+EC behaves much like RAID 5 or 6...for *k* blocks of data, *m* blocks of
+parity data are generated, from which the original data can be recovered in the
+event of disk or node failures (erasures, in EC parlance).  A typical EC scheme to use is Reed-Solomon 6-3, where
+6 blocks of data produce 3 blocks of parity, an overhead of only 50%.  In addition
+to the factor of 2 increase in available disk space, RS-6-3 is also more fault
+tolerant...a loss of 3 data blocks can be tolerated, compared to triple replication
+where only two blocks can be lost.
+
+More storage, better resiliency, so what's the catch?  One concern with using EC is
+the time spent calculating the parity blocks.  Unlike the default replication write
+path, where a client writes a block, and then the DataNodes take care of replicating
+the data, an EC HDFS client is responsible for computing the parity and sending that
+to the DataNodes.  This increases the CPU and network load on the client.  The CPU
+hit can be mitigated through the use of the Intel ISA-L library, but only on CPUs
```

Comment:
yes, I believe it would.  I think the native EC libs can make use of the ISA-L library.  This brings up a point I'd forgotten...to get the full benefit of the ISA-L acceleration, all clients need visibility to the libisal.so library.  I think in my early testing the tablet servers weren't set up correctly, so they didn't get the acceleration.  That might be a little in the weeds for the purposes of this post, though.  Would be a good mention in the documentation if an EC section is added

Reply:
If tserver specific settings are needed to make the ISA-L library visible it would be nice to mention those somewhere.  The HOWTO you mentioned would be ideal.